streams/runners/storm/runtime/src/main/java/com/hortonworks/streamline/streams/runtime/storm/spout/AvroStreamsSnapshotDeserializer.java

```diff
@@ -65,7 +65,10 @@ private Object convertValue(Object deserializedObj) {
             List<Schema.Field> fields = indexedRecord.getSchema().getFields();
             ImmutableMap.Builder<String, Object> keyValues = ImmutableMap.builder();
             for (Schema.Field field : fields) {
-                keyValues.put(field.name(), convertValue(indexedRecord.get(field.pos())));
+                Object currentValue = convertValue(indexedRecord.get(field.pos()));
+                if (currentValue != null) {
```

Comment:
@harshach If we want StreamlineEvent to contain all keys(even for null values) then we have to go with the route suggested by Satish. @satishd I have code changes like you suggested but we will run in to issue further down. StreamlineEventImpl uses ImmutableMap and when an instance of StreamlineEventImpl is constructed it will try to create an ImmutableMap again with an entry that has a null for its value. https://github.com/hortonworks/streamline/blob/master/streams/common/src/main/java/com/hortonworks/streamline/streams/common/StreamlineEventImpl.java#L105 is the line where it will throw an Exception now. I will need to change StreamlineEventImpl as well to not use ImmutableMap since it does not allow null values and it will need more testing.